tiledb/sm/serialization/fragment_metadata.cc

```diff
@@ -231,7 +252,8 @@ Status fragment_metadata_from_capnp(
     // If there are no levels, we still need domain_ properly initialized
     frag_meta->rtree() = RTree(&domain, constants::rtree_fanout);
     Deserializer deserializer(data.begin(), data.size());
-    frag_meta->rtree().deserialize(deserializer, &domain, frag_meta->version());
+    frag_meta->rtree().deserialize(
+        deserializer, &domain, constants::format_version);
```

Comment:
Can you please explain if there is a reason why we use constants::format_version here and not the deserialized version number?

Reply:
The reason is that what we actually deserialize is not something written on disk in a possibly historical format, but what has been serialized in `fragment_metadata_to_capnp` using `frag_meta.rtree().serialize(serializer)`. This means that no matter what the version of a fragment is on disk, we will be serializing _on wire_ in fragment_metadata_to_capnp in the most "modern" way, so we need to deserialize it as well in that way. Let's discuss offline if this is still unclear, I am adding a comment also inline because this is indeed very confusing - I discoved it during back-comp testing.